### ISSUE TITLE
Unclear error message when assigning wrong type to polymorphic relationship

### ISSUE THREAD

**Issue by balinterdi** [emberjs/data#4633]
I have the following relationships:

```js
// app/models/business.js
import DS from 'ember-data';

const { attr, hasMany, belongsTo, Model } = DS;

export default Model.extend({
  logo:  belongsTo('image', { async: false }),
});
```

```js
// app/models/image.js
import DS from 'ember-data';

const { Model, attr, belongsTo } = DS;

export default Model.extend({
  pictured:  belongsTo('pictured', { polymorphic: true })
});
```

When a logo.pictured relationship is set, I get the following error:

> You cannot add a record of type 'business' to the 'image.pictured' relationship (only 'pictured' allowed)

I think this error might be made clearer as [at the point it is thrown](https://github.com/emberjs/data/blob/234042672fc74af5ddef3dd8f517eebb99267295/addon/-private/debug.js#L72) we know we're checking a polymorphic relationship so we might as well say:

> "The `business` type does not implement `pictured` and thus cannot be assigned to the `pictured` relationship in `image`. Make it a descendant of `pictured` or use a mixin of the same name.

What do you think?

(Showed this to @pangratz at EmberFest a few days ago.)

**Comment by runspired** [comment]
@balinterdi I'd be all for this :)

**Comment by pete-the-pete** [comment]
@runspired and @balinterdi does this still need a contributor? If so, I'd like to pick it up.

**Comment by runspired** [comment]
@pete-the-pete it does!

**Comment by balinterdi** [comment]
@pete-the-pete Yes, feel free to snag it. Thank you!

**Comment by runspired** [comment]
Resolved by #5518
